polars/polars-io/src/csv/write_impl.rs

```diff
@@ -95,12 +98,17 @@ fn write_anyvalue(
                 TimeUnit::Microseconds => temporal_conversions::timestamp_us_to_datetime(v),
                 TimeUnit::Milliseconds => temporal_conversions::timestamp_ms_to_datetime(v),
             };
-            match tz {
-                None => write!(f, "{}", ndt.format(datetime_format)),
-                Some(tz) => {
-                    write!(f, "{}", PlTzAware::new(ndt, tz))
-                }
-            }
+            let formatted = match tz {
+                Some(tz) => match tz.parse::<Tz>() {
+                    Ok(parsed_tz) => parsed_tz.from_utc_datetime(&ndt).format(datetime_format),
+                    Err(_) => match temporal_conversions::parse_offset(tz) {
+                        Ok(parsed_tz) => parsed_tz.from_utc_datetime(&ndt).format(datetime_format),
+                        Err(_) => unreachable!(),
+                    },
+                },
+                _ => ndt.format(datetime_format),
+            };
+            write!(f, "{formatted}")
```

Comment:
Hi @alexander-beedie - do you have any thoughts on this? On the one hand, parsing `tz` for each element slows things down for tz-aware columns, and could be done beforehand in a per-column fashion

On the other hand, in https://github.com/pola-rs/polars/pull/4724 it looks like you intentionally tried to avoid per-column inference

Reply:
Ahh, yes - though it wasn't about not wanting per-column inference (which would actually be great) as ensuring that any such inference was done outside the hot loop (per-_element_ inference would be bad). 

I took a minimal approach in the end as my initial attempt to reshuffle things on a per-column basis became unnecessarily convoluted - treat my earlier commit as a mere first step towards a more flexible/better per-column future... ;)